### PR TITLE
handle copy of live photos

### PR DESCRIPTION
**Pull request by ArtificialOwl**
- uses of handleMove() during the `BeforeNodeCopiedEvent` to check the doability of copying both files.
- handleCopy() called during the `NodeCopiedEvent` to generate a copy of the linked file.

### CODE REVIEW

File: apps/files/lib/Listener/SyncLivePhotosListener.php [hunk 1]
```diff
@@ -98,14 +105,13 @@ public function handle(Event $event): void {
 	 * of pending renames inside the 'pendingRenames' property,
 	 * to prevent infinite recursive.
 	 */
-	private function handleMove(BeforeNodeRenamedEvent $event, Node $peerFile): void {
+	private function handleMove(Event $event, Node $peerFile, bool $prepForCopyOnly = false): void {
```
Comment (artonge) [inline apps/files/lib/Listener/SyncLivePhotosListener.php:108]: Or extract the logic into a separate method?

Comment (artonge) [inline apps/files/lib/Listener/SyncLivePhotosListener.php:108]:
```suggestion
	private function handleMove(AbstractNodesEvent $event, Node $peerFile, bool $prepForCopyOnly = false): void {
```

File: apps/files/lib/Listener/SyncLivePhotosListener.php [hunk 2]
```diff
@@ -129,15 +135,47 @@ private function handleMove(BeforeNodeRenamedEvent $event, Node $peerFile): void
 			return;
 		}
 
-		$this->pendingRenames[$sourceFile->getId()] = $targetPath;
+		$this->pendingRenames[$sourceFile->getId()] = $peerFile->getId();
```
Comment (artonge) [inline apps/files/lib/Listener/SyncLivePhotosListener.php:138]: Could even be a simple boolean as we do not use the value in the end.

**Review by artonge — APPROVED** [review]

File: lib/public/Files/Events/Node/BeforeNodeCopiedEvent.php [hunk 3]
```diff
@@ -25,8 +25,31 @@
...
+use OCP\Files\Node;
+
 /**
  * @since 20.0.0
  */
 class BeforeNodeCopiedEvent extends AbstractNodesEvent {
+	/**
+	 * @since 20.0.0
+	 */
+	public function __construct(Node $source, Node $target, private bool &$run) {
```
[the remark this answers is not in the document]
Comment (artonge) [reply lib/public/Files/Events/Node/BeforeNodeCopiedEvent.php:38]: How can the listener gives the new value of `$run` back to the emitter?

[the remark this answers is not in the document]
Comment (icewind1991) [reply lib/public/Files/Events/Node/BeforeNodeCopiedEvent.php:38]: add a getter so `HookConnector` can retrieve it

[the remark this answers is not in the document]
Comment (artonge) [reply lib/public/Files/Events/Node/BeforeNodeCopiedEvent.php:38]: That sounds smart.  @ArtificialOwl will you have the time to update the other event listeners too ? :D

[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/public/Files/Events/Node/BeforeNodeCopiedEvent.php]: @ArtificialOwl :eyes:

[the remark this answers is not in the document]
Comment (ArtificialOwl) [reply lib/public/Files/Events/Node/BeforeNodeCopiedEvent.php]: Tested few other ways but the simplest seems to have the HookController catching new event `AbortedEventException` and update `$arguments['run']`. Throwing the exception also stop the event propagation.

@icewind1991 @artonge please re-review

File: lib/public/Files/Events/Node/BeforeNodeDeletedEvent.php [hunk 4]
```diff
@@ -25,31 +25,8 @@
...
-	 */
-	public function __construct(Node $node, private bool &$run) {
-		parent::__construct($node);
-	}
-
-	/**
-	 * @since 28.0.0
-	 * @return never
-	 */
-	public function abortOperation(\Throwable $ex = null) {
```
Comment (icewind1991) [inline lib/public/Files/Events/Node/BeforeNodeDeletedEvent.php:46]: This is part of the public api so we can't remove it without a deprecation period.

You could change it to just throw the `AbortedEventException`

File: lib/public/Files/Events/Node/BeforeNodeRenamedEvent.php [hunk 5]
```diff
@@ -25,31 +25,8 @@
...
-	 */
-	public function __construct(Node $source, Node $target, private bool &$run) {
-		parent::__construct($source, $target);
-	}
-
-	/**
-	 * @since 28.0.0
-	 * @return never
-	 */
-	public function abortOperation(\Throwable $ex = null) {
```
Comment (icewind1991) [inline lib/public/Files/Events/Node/BeforeNodeRenamedEvent.php:46]: same here

**Review by icewind1991 — CHANGES_REQUESTED** [review]

**Review by icewind1991 — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Comment by ArtificialOwl** [comment]
@artonge based on the size of the changes and the need for this feature, do we want to backport to 28 ?

**Comment by artonge** [comment]
I would say no, but not for me to decide, @sorbaugh

**Comment by squid-f** [comment]
Hi. Just updated to 29.0.3 and the error persists : 
` AbortedEventException Cannot delete the video part of a live photo
delete process aborted `

**Pull request merged**